src/constants/TokenConstantsLookupTable.ts

```diff
@@ -0,0 +1,58 @@
+import store from 'state'
+
+import { SupportedChainId } from './chains'
+import { DEFAULT_LIST_OF_LISTS } from './lists'
+
+type lookupKey = `0x${string}:${SupportedChainId}`
+interface TokenConstants {
+  decimals?: number // uint8
+  name?: string
+  symbol?: string
+  chainChecked: boolean
+}
+class TokenConstantsLookupTable {
+  private dict: Record<lookupKey, TokenConstants | undefined> = {}
+  private initialized = false
+
+  initialize() {
```

Comment:
when will this init? how do you ensure the store has been properly hydrated with the lists?